### PR TITLE
One patch to align - Testsuite for standard interpreter - Dynarec lin…

### PR DESCRIPTION
**Pull request by FCare**
…ux x64 support - New Vsync mechanism - New Async scsp synchro

Here is the updated patch without the build directory

### CODE REVIEW

File: yabause/src/sh2int.c [hunk 1]
```diff
@@ -427,21 +427,21 @@ static void FASTCALL SH2bf(SH2_struct * sh)
...
       sh->cycles += 2;
-      SH2delay(sh, temp + 2);
    }
    else
    {
-      sh->regs.PC += 2;
+      sh->regs.PC += 4;
       sh->cycles++;
    }
+   SH2delay(sh, temp + 2);
```
Comment (devmiyax) [inline yabause/src/sh2int.c]: What game is improved with this change?

Check this document P.140.  When T=0 delay slot is not needed.
http://www.embeddeddeveloper.com/cores/documents/superhSM.pdf

File: yabause/src/sh2int.c [hunk 2]
```diff
@@ -252,8 +252,10 @@ static void FASTCALL SH2delay(SH2_struct * sh, u32 addr)
 #endif
 
    // Execute it
-   sh->regs.PC -= 2;
+   u32 tmp = sh->regs.PC;
+   sh->regs.PC = addr;
    opcodes[sh->instruction](sh);
+   sh->regs.PC = tmp;
```
Comment (devmiyax) [inline yabause/src/sh2int.c]: This is the fix of #268 .
https://github.com/devmiyax/yabause/commit/d8122116942c6326b7d508270fd6818fb014ab92

File: yabause/src/sh2int.c [hunk 3]
```diff
@@ -252,8 +252,10 @@ static void FASTCALL SH2delay(SH2_struct * sh, u32 addr)
 #endif
 
    // Execute it
-   sh->regs.PC -= 2;
+   u32 tmp = sh->regs.PC;
```
Comment (devmiyax) [inline yabause/src/sh2int.c]: This is the fix of #268 .
https://github.com/devmiyax/yabause/commit/d8122116942c6326b7d508270fd6818fb014ab92

File: yabause/src/sh2int.c (hunk 1 above)
Comment (FCare) [inline yabause/src/sh2int.c]: Gee, this doc does not give the same behavior as http://www.shared-ptr.com/sh_insns.html

Technically, it is not changing anything. You do not need to take this change. Or I can revert if you prefer. As I changed the PC+=2 to PC+=4, then it is the same functionnality.

File: yabause/src/sh2int.c (hunk 3 above)
Comment (devmiyax) [inline yabause/src/sh2int.c]: please check this

**Comment by devmiyax** [comment]
Regression on interpreter mode 32bit.

image

**Comment by BenjaminSiskoo** [comment]
@Devmiayx is it he same kind of problem than UoYabause 0.5.2 (your fork) ?

**Comment by FCare** [comment]
Yep, it seems your patch 0f26a7567aeb87ead7e9169394886e691b78bc89 [Tomb Raider (E)] Polygons problems and black band.... #356 broke a lot of stuff...

**Comment by devmiyax** [comment]
No there are no polygons on this game.

**Comment by devmiyax** [comment]
Sorry for that. I was testing on software rendering mode.  On OPenGL rendering mode it works fine.

**Comment by FCare** [comment]
Ok, thanks for clarification. I personnaly doubt that the dynarec is
working on 32bits. I saw some parts are missing. I will fix for 32 bits on
linux and on Windows (different calling coventions)

**Comment by devmiyax** [comment]
Fail to build when targeting Android.

C:/ext/GitHub/yabause_hv/yabause/src/thr-linux.c: At top level:
C:/ext/GitHub/yabause_hv/yabause/src/thr-linux.c:263:3: error: unknown type name 'pthread_barrier_t'
   pthread_barrier_t barrier;

**Comment by FCare** [comment]
As the pull has not been accepted earlier, it is now too far from the android build.
Android support is not in my objective. As I will not fix it, I am closing the pull request.

**Pull request closed without merging**